torch/csrc/api/include/torch/data/dataloader.h

```diff
@@ -239,6 +263,20 @@ class DataLoader {
     return torch::make_unique<detail::sequencers::NoSequencer<Result>>();
   }
 
+  /// Called by a worker thread when its dataset indicates exhaustion (for
+  /// ExhaustibleDataset).
+  void indicate_worker_dataset_is_exhausted() {
+    // If this is the last worker thread alive, we need to indicate to the main
+    // thread that it should now die. We do this by pushing a `Quit` result into
+    // the result queue. The `push_job` and `pop_job` are required by the
+    // contract of the datashuttle.
+    if (--live_worker_count_ == 0) {
```

Comment:
live_worker_cound_ is not necessarily always equal to the active worker thread count. While one worker is actively fetching batch data, it is possible that another worker thread is exhausted, returns empty result, asks for a new get_batch job and returns empty result again. In this case, indicate_worker_dataset_is_exhausted() is called twice,  live_worker_count_ is decreased twice due to the same thread.